service/storage_service.cc

```diff
@@ -1074,6 +1067,51 @@ class topology_coordinator {
         co_return std::pair{gen_uuid, std::move(gen_mutations)};
     }
 
+    // Broadcasts all mutations returned from prepare_new_cdc_generation_data except the last one.
+    // Each mutation is sent in separate raft command.
+    // It takes group0_guard and if the number of mutations is greater than one, the guard is dropped
+    // and a new one is created and returned, otherwise the old one will be returned.
+    // Commands are sent sequentially and unguarded (guard used for sending the last mutation will
+    // guarantee that the term hasn't been changed).
+    // Returns the generation's UUID, guard and last mutation which
+    // will be sent with additional topology data by the caller.
+    //
+    // With the default commitlog segment size, `mutation_size_threshold` will be 4 MB. In large clusters e.g.
+    // 100 nodes, 64 shards per node, 256 vnodes cdc generation data can reach the size of 30 MB, thus
+    // there will be no more than 8 commands.
+    //
+    // In a multi-DC cluster with 100ms latencies between DCs, which means ~200ms for each command,
+    // because we first perform a barrier in start_operation and then commit the command. Thus this operation will
+    // take about 1.6s which is incomparably smaller than bootstrapping operation.
+    future<std::tuple<utils::UUID, group0_guard, canonical_mutation>> prepare_and_broadcast_cdc_generation_data(
+            locator::token_metadata_ptr tmptr, group0_guard guard, std::optional<bootstrapping_info> binfo) {
+        auto [gen_uuid, gen_mutations] = co_await prepare_new_cdc_generation_data(tmptr, guard, binfo);
+
+        if (gen_mutations.empty()) {
+            on_internal_error(slogger, "cdc_generation_data: gen_mutations is empty");
+        }
+
+        std::vector<canonical_mutation> updates{gen_mutations.begin(), gen_mutations.end()};
+
+        if (updates.size() > 1) {
+            guard.~group0_guard();
+
+            for (size_t i = 0; i < updates.size() - 1; ++i) {
+                auto const reason = format(
+                    "insert CDC generation data (UUID: {}), part {}/{}", gen_uuid, i + 1, updates.size());
+
+                slogger.trace("raft topology: do update {} reason {}", updates[i], reason);
+                write_mutations change{{std::move(updates[i])}};
+                group0_command g0_cmd = _group0.client().prepare_command(std::move(change));
+                co_await _group0.client().add_entry_unguarded(std::move(g0_cmd));
```

Comment:
Better to check that term did not change between iteration to not do useless work.

Reply:
`write_mutations` are sent in parallel now, so checking that the term did not change is pointless.